nodes/generator/torusKnot.py

```diff
@@ -0,0 +1,432 @@
+    # NOTE: linkIndex value is in [0, links-1]
+    linkPhase = 2*pi/q * linkIndex  # = 0 when there is just ONE link
+
+    # user defined phasing
+    if not options_plus:
+        rPhase = 0
+        sPhase = 0
+
+    rPhase += linkPhase  # total revolution phase of the current link
+
+    # flip directions ? NOTE: flipping both is equivalent to no flip
+    if flipP:
+        p *= -1
+    if flipQ:
+        q *= -1
+
+    # create the list of verts, edges and norms for the current link
+    listVerts = []
+    listEdges = []
+    listNorms = []
+    for n in range(N-1):
+        # t = 2*pi / links * n/(N-1) with: da = 2*pi/links/(N-1) => t = n * da
+        t = n * da
+        theta = p*t*u + rPhase  # revolution angle
+        phi   = q*t*v + sPhase  # spin angle
+
+        # cache values to improve performance
+        sin_theta = sin(theta)
+        cos_theta = cos(theta)
+        sin_phi = sin(phi)
+        cos_phi = cos(phi)
+
+        # compute vertex coordinates
+        x = (R + r*cos_phi) * cos_theta
+        y = (R + r*cos_phi) * sin_theta
+        z = r*sin_phi * h
+
+        # append vertex
+        listVerts.append([x, y, z])
+
+        # append edge
+        if n != N-2:
+            listEdges.append([n, n+1])
+        else: # closing the loop
+            listEdges.append([n, 0])
+
+        # append normal
+        cx = R * cos_theta # torus tube center at theta
+        cy = R * sin_theta # torus tube center at theta
+        norm = [x-cx, y-cy, z]
+        listNorms.append(norm)
+
+    if DEBUG:
+        end = time.time()
+        print("Torus Link Time: ", end - start)
+
+    return listVerts, listEdges, listNorms
+
+
+class SvTorusKnotNode(bpy.types.Node, SverchCustomTreeNode):
+    ''' Torus Knot '''
+    bl_idname = 'SvTorusKnotNode'
+    bl_label = 'Torus Knot'
+    bl_icon = 'MESH_TORUS'
+
+    # switch radii input sockets (R,r) <=> (eR,iR)
+    def update_mode(self, context):
+        if self.mode == 'EXT_INT':
+            self.inputs['R'].prop_name = "torus_eR"
+            self.inputs['r'].prop_name = "torus_iR"
+        else:
+            self.inputs['R'].prop_name = "torus_R"
+            self.inputs['r'].prop_name = "torus_r"
+
+        updateNode(self, context)
+
+    # keep the equivalent radii pair in sync (eR,iR) => (R,r)
+    def external_internal_radii_changed(self, context):
+        if self.mode == "EXT_INT":
+            self.torus_R = (self.torus_eR + self.torus_iR)*0.5
+            self.torus_r = (self.torus_eR - self.torus_iR)*0.5
+            updateNode(self, context)
+
+    # keep the equivalent radii pair in sync (R,r) => (eR,iR)
+    def major_minor_radii_changed(self, context):
+        if self.mode == "MAJOR_MINOR":
+            self.torus_eR = self.torus_R + self.torus_r
+            self.torus_iR = self.torus_R - self.torus_r
+            updateNode(self, context)
+
+    # GENERAL options
+    options_plus = BoolProperty(
```

Comment:
I did not notice that :) Good catch. I will keep that in mind and perhaps change it to 4 in a future update.